src/components/Timeline/TimelineUpdate.tsx

```diff
@@ -1,95 +1,27 @@
-import { FormattedMessage } from 'react-intl';
-import { Box, Grid, Typography } from '@material-ui/core';
-
-import PersonHoverCard from 'components/PersonHoverCard';
-import ZetkinPerson from 'components/ZetkinPerson';
-import ZetkinRelativeTime from 'components/ZetkinRelativeTime';
-import { ZetkinPerson as ZetkinPersonType, ZetkinUpdate } from 'types/zetkin';
-
-const getPersonName = (person: Partial<ZetkinPersonType>) =>
-  person.first_name + ' ' + person.last_name;
+import TimelineAssigned from './updates/TimelineAssigned';
+import TimelineJourneyMilestone from './updates/TimelineJourneyMilestone';
+import {
+  UPDATE_TYPES,
+  ZetkinUpdate,
+  ZetkinUpdateAssignee,
+  ZetkinUpdateJourneyMilestone,
+} from 'types/updates';
 
 interface Props {
   update: ZetkinUpdate;
 }
 
 const TimelineUpdate: React.FunctionComponent<Props> = ({ update }) => {
-  return (
-    <Box display="flex" flexDirection="column">
-      <Grid alignItems="center" container direction="row" spacing={1}>
-        {update.actor && renderActorAvatar(update.actor as ZetkinPersonType)}
-        {renderDescriptionText()}
-      </Grid>
-      {getContent()}
-    </Box>
-  );
-
-  // optional "actor" (the person who made the change)
-  function renderActorAvatar(actor: ZetkinPersonType) {
-    return (
-      <>
-        <Grid item>
-          <PersonHoverCard personId={actor.id}>
-            <ZetkinPerson id={actor.id} name={''} showText={false} small />
-          </PersonHoverCard>
-        </Grid>
-      </>
-    );
-  }
-
-  // Forms the update description, which consists of an "action" (i.e. what the update changed),
-  function renderDescriptionText() {
+  if (update.type === UPDATE_TYPES.JOURNEYINSTANCE_ADDASSIGNEE) {
+    return <TimelineAssigned update={update as ZetkinUpdateAssignee} />;
```

Comment:
Is there a way to make this inferred by the assertion above so it can be done without typecasting? If it's too much work, ignore it because we can be sure with the types that it will work fine, but it's nice to avoid `as` when possible.

Reply:
Not much else to say about this except: If I could see a way to do that, I would have done it already

Reply:
It would be possible by having different types for each update type, and hard-coding the `type` attribute in them using that enum, e.g.:

<img width="457" alt="image" src="https://user-images.githubusercontent.com/550212/165538069-7e8c3989-a467-4a87-b7bb-7fe96795e129.png">

Note how the concrete type of `update` is known within that `if` statement, because the condition will only be true for `ZetkinUpdateAssignee`.

I think this is generally a nice way to solve things because you get 100% type safety, and with generics it could be very efficient in terms of lines of code. But that's a refactor we can make later if you are more comfortable working this way, @lowlandjuju.